### PR TITLE
Adds image button to wiki Editor

### LINKED ISSUE #8241: Add a button for uploading images to the Wiki Editor toolbar that would look similar to the Rich Editor's button
To add images:
prompt2

Toolbar iteration 1:
tool

### LINKED ISSUE #8259: Re-order legacy and comment toolbars so order of tools matches that of Rich Text Editor
Currently:
current

Toolbar iterations including the image icon:
1.
t1

2.
t2

3.
t3

4.
t4

5.
t5

6.
t6

### PR DESCRIPTION
**Pull request by shreyaa-sharmaa**
Fixes #8241, fixes #8259 

simplescreenrecorder-2020-08-08_13 22 07

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Exciting! Actually i wonder if we could do something a little simpler, i left a comment here! https://github.com/publiclab/plots2/issues/8241#issuecomment-670007797

Thank you so much @shreyaa-sharmaa !!

**Comment by jywarren** [comment]
I'm thinking it could be related to @NitinBhasneria's work here! https://github.com/publiclab/plots2/pull/8240

**Comment by shreyaa-sharmaa** [comment]
@jywarren its done. :tada:

**Comment by shreyaa-sharmaa** [comment]
I've also made a small changes to the toolbar so we can combine it with #8259.

**Comment by shreyaa-sharmaa** [comment]
@sagarpreet-chadha I've added the condition for file upload. Should I change it to `accept="image/*"` instead?

We can upload one file at a time (similar to how we did it before) and thus the selected picture would come one after the other based on order of addition.

**Comment by jywarren** [comment]
This sounds like a great plan, @shreyaa-sharmaa ! Let's do it!

**Comment by shreyaa-sharmaa** [comment]
This is not ready to be merger yet.

**Comment by jywarren** [comment]
OK, no rush, just tell us when you'd like to get this added in! Much appreciated!!

File: app/assets/stylesheets/editor.css [hunk 1]
```diff
@@ -160,39 +160,10 @@ version of the editor is released. */
   resize: vertical;
 }
 
-#imagebar {
```
Comment (jywarren) [inline app/assets/stylesheets/editor.css:163]: Hi @shreyaa-sharmaa -- actually, we'd like to keep the image upload bar as well, so we have both. Will that be possible? Thank you! I'm sorry i missed this.

Comment (shreyaa-sharmaa) [reply app/assets/stylesheets/editor.css:163]: Will do that.

**Comment by shreyaa-sharmaa** [comment]
This is almost done. I just need to ask about the placement of the progress bar. Should we have a progress bar below the textarea or to the right of it (same as that of main image)? 
cc: @jywarren @sagarpreet-chadha

**Comment by sagarpreet-chadha** [comment]
Let me ask UX experts @jywarren @ebarry for this, thanks 💯  Great work so far 😄

**Comment by jywarren** [comment]
Hi! I believe we do it this way in the comment boxes... is it easy to just use the code that drives this?

Thank you!!!

Screen Shot 2020-08-25 at 1 07 24 PM

**Comment by ebarry** [comment]
The progress bar as @jywarren advises looks good! 
For reference, here's what the progress bar looks like on /post : 
Untitled

**Comment by jywarren** [comment]
I believe this is for inline image additions, not for the lead image, though, which is what @ebarry is highlighting. Thanks though, it's good context!

**Comment by shreyaa-sharmaa** [comment]
The original code for progress-bar for inline images doesn't work. I tried to use one similar to the comment boxes (noting here that the same function triggers progress bar in both these cases) but that didn't work as well. Right now I'm tracking it down. I'll hopefully get it done by tomorrow. Anyone here has any idea why the original one doesn't work? :confused:

**Comment by jywarren** [comment]
Hmm, i'm sorry you're having trouble! The original code presumably works in a couple other contexts still, like the comment box. I think it was complicated because it was intended to be re-used code for various contexts. Where is it failing? Could it be due to css selectors not being specific enough?

File: app/views/editor/_editor.html.erb [hunk 2]
```diff
@@ -3,32 +3,10 @@
 <div class="form-group dropzone">
   <textarea aria-label="Wiki Text" name="body" tabindex="2" class="form-control" id="text-input" rows="20" cols="60"><% if @node && @node.latest && @node.latest.body %><%= @node.latest.body %><% else %><%= params[:body] %><% end %></textarea>
   <div id="imagebar">
-    <div style="display:none;" id="create_progress" class="progress progress-striped active pull-right">
-      <div id="create_progress-bar" class="progress-bar" style="width: 0%;"></div>
+    <div id="create_progress" style="display:none;width:100%;" class="progress progress-striped active">
+      <div class="create_progress-bar" role="progressbar" style="width: 0%;"></div>
     </div>
-    <p>
```
Comment (jywarren) [inline app/views/editor/_editor.html.erb:9]: Perhaps we have to make this more accurately match working upload image bars like in comments, to get it to work again?

Comment (jywarren) [reply app/views/editor/_editor.html.erb:9]: For example:

https://github.com/publiclab/plots2/blob/c229df48ffca69038f5b26f5cda2baa56ee94634/app/views/comments/_form.html.erb#L48-L76

Comment (jywarren) [reply app/views/editor/_editor.html.erb:9]: Actually, see for reference @NitinBhasneria's work here! https://github.com/publiclab/plots2/pull/8240/files Just so you can see how the system test walks us through the process with some working code.

**Comment by shreyaa-sharmaa** [comment]
It's working now. GIF of changes:
simplescreenrecorder-2020-09-05_12 32 03

Please test it in gitpod and let me know if any changes are required.

File: app/assets/javascripts/dragdrop.js [hunk 3]
```diff
@@ -51,13 +51,19 @@ jQuery(document).ready(function() {
...
+                $('#imagebar .inline_image_drop').show();
+                $('#create_progress').show();
+                $('#create_uploading').show();
                 elem = $($D.selected).closest('div.comment-form-wrapper').eq(0);
                 elem.find('#create_progress').eq(0).show();
                 elem.find('#create_uploading').eq(0).show();
                 elem.find('#create_prompt').eq(0).hide();
                 elem.find('.dropzone').eq(0).removeClass('hover');
             },
             done: function (e, data) {
```
Comment (shreyaa-sharmaa) [inline app/assets/javascripts/dragdrop.js:63]: This PR has a codeclimate issue. The `done` function is of 27 lines whereas the limit is 25. How can it be optimized further?

Comment (sagarpreet-chadha) [reply app/assets/javascripts/dragdrop.js:63]: Remove blank line 73

Comment (sagarpreet-chadha) [reply app/assets/javascripts/dragdrop.js:63]: Declare multiple variables in one line.

**Comment by shreyaa-sharmaa** [comment]
I believe this is complete now. Please have a look @jywarren @sagarpreet-chadha

**Review by jywarren — APPROVED** [review]
This looks fantastic. Thank you so much, @shreyaa-sharmaa for bringing this to the finish line!!! 👍 🎉

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Screen Shot 2020-09-15 at 12 21 16 PM

! Nice!!!

**Pull request merged**